share/cmake/cccl/cccl-config.cmake

```diff
@@ -0,0 +1,59 @@
+#
+# find_package(CCCL) config file.
+#
+# Imports the Thrust, CUB, and libcudacxx components of the NVIDIA
+# CUDA/C++ Core Libraries.
+
+unset(cccl_version) # TODO
+
+set(cccl_cmake_dir "${CMAKE_CURRENT_LIST_DIR}")
+
+if (${CMAKE_FIND_PACKAGE_NAME}_FIND_QUIETLY)
+  set(cccl_quiet_flag "QUIET")
+else()
+  set(cccl_quiet_flag "")
+endif()
+
+foreach(component ${${CMAKE_FIND_PACKAGE_NAME}_FIND_COMPONENTS})
+  unset(req)
+  if (${CMAKE_FIND_PACKAGE_NAME}_FIND_REQUIRED_${component})
+    set(cccl_comp_required_flag "REQUIRED")
+  endif()
+
+  if(component STREQUAL "libcudacxx")
```

Comment:
Instead of iterating over the `FIND_COMPONENTS` we can remove the iteration and instead do
`if("libcudacxx" IN_LIST ${CMAKE_FIND_PACKAGE_NAME}_FIND_COMPONENTS)`

Reply:
Ended up keeping the loop, since we need to lookup the per-component required flags using the user-specified strings.